provider/direct.go

```diff
@@ -37,21 +47,48 @@ func (f CredCheckerFunc) Check(user, password string) (ok bool, err error) {
 	return f(user, password)
 }
 
+// Credentials holds user credentials
+type Credentials struct {
+	User     string `json:"user"`
+	Password string `json:"passwd"`
+	Audience string `json:"aud"`
+}
+
 // Name of the handler
 func (p DirectHandler) Name() string { return p.ProviderName }
 
-// LoginHandler checks "user" and "passwd" against data store and makes jwt if all passed
-// GET /something?user=name&password=xyz&sess=[0|1]
+// LoginHandler checks "user" and "passwd" against data store and makes jwt if all passed.
+//
+// GET /something?user=name&passwd=xyz&aud=bar&sess=[0|1]
+//
+// POST /something?sess[0|1]
+// Accepts application/x-www-form-urlencoded or application/json encoded requests.
+//
+// application/x-www-form-urlencoded body example:
+// user=name&passwd=xyz&aud=bar
+//
+// application/json body example:
+// {
+//   "user": "name",
+//   "passwd": "xyz",
+//   "aud": "bar",
+// }
 func (p DirectHandler) LoginHandler(w http.ResponseWriter, r *http.Request) {
-	user, password := r.URL.Query().Get("user"), r.URL.Query().Get("passwd")
-	aud := r.URL.Query().Get("aud")
+	if r.Body != nil {
```

Comment:
probably, moving this body wrapping to getCredentials will make this part closer to the consumer and more readable